lib/raidenclient/RaidenClient.ts

```diff
@@ -331,6 +331,11 @@ class RaidenClient extends SwapClient {
     }
   }
 
+  public canRouteToNode = async (destination: string, currency: string) => {
+    const route = await this.getRoute(1, destination, currency);
```

Comment:
I think you missed `peer.raidenAddress && swapClient.isConnected()` from previous logic here.

Reply:
The check for `peer.raidenAddress` is done in `canRouteToPeer` when we call `peer.getIdentifier` and return right away if the result is undefined. `getIdentifer` will return the raiden address when it's called with a raiden swapClient.

As for checking `isConnected`, I worked on this PR right after #1398, which ensures we are connected before even attempting this call. You're right that it's missing in this PR and I should have kept it or at least made note of it here, in my mind I was taking the changes for granted. But now that the other one is merged I don't think it's necessary.

However, in thinking about this some more there's still a possibility for this call to throw an unhandled exception even with the other PR's changes. We can suddenly become disconnected from the client and make this call before the status updates, or we can hit some other unexpected exception. So what I'm going to do is wrap the `canRouteToNode` call with a try/catch that returns `false` in case of an exception - the logic being that if we can't make this check then we can't route payments in the current state.